src/battle-text-parser.ts

```diff
@@ -826,8 +826,8 @@ class BattleTextParser {
 		}
 
 		case '-immune': {
-			let [, pokemon, effect] = args;
-			const line1 = this.maybeAbility(effect, kwArgs.of || pokemon);
+			const [, pokemon] = args;
+			const line1 = this.maybeAbility(kwArgs.from, kwArgs.of || pokemon);
 			let template = this.template('block', effect);
```

Comment:
Although maybe this should be `kwArgs.from` too?

I don't have replays so I can't see this.

Reply:
You're right of course, I notice that for example Own Tempo needs it. Also sorry but I did this very late at night so I didn't stay around to watch for the test failure myself. And thanks for the tip about `build full`, it only takes half a minute on my machine so that won't be a problem.

Reply:
You don't need to do it anymore (I moved text resource files out of `graphics.js`), and should probably actually delete `data/graphics.js` from your computer to speed up build times.